scripts/utils/docs-content-tools/create-nav-yaml.js

```diff
@@ -0,0 +1,159 @@
+const fs = require('fs');
+const dirTree = require('directory-tree');
+const yaml = require('js-yaml');
+
+const WRITE_DIRECTORY = './src/docs-content-nav/';
+const READ_DIRECTORY = './src/content/docs/';
+
+/**
+ * Strip the file extension from the path of the file
+ * @param {String} path The path to strip the extension from.
+ * @returns {String} The path without the file extension.
+ */
+const stripPath = (path) =>
+  path.endsWith('.mdx') ? path.substring(0, path.lastIndexOf('.')) : path;
+
+/**
+ * Get the slug for the path
+ * @param {String} path The path to get the slug for.
+ * @returns {String} The slug for the path.
+ */
+const getSlug = (path) => {
+  const strippedPath = stripPath(path);
+  return strippedPath.match(/content(.*)/)[1];
+};
+
+/**
+ * Convert filename with - to readable title
+ * @param {String} name The filename to convert to a title.
+ * @returns {String} The title.
+ */
+const convertToTitle = (name) => {
+  const strippedPath = stripPath(name);
+  return strippedPath
+    .split('-')
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ');
+};
```

Comment:
we actually need to do both because the titles that are directories won't have frontmatter, so i've added a list of special words that can be added to